src/main/java/com/theelixrlabs/healthcare/dao/DoctorPatientAssignmentRepository.java

```diff
@@ -18,5 +19,5 @@ public interface DoctorPatientAssignmentRepository extends MongoRepository<Docto
      * @param patientID    UUID of the patient associated with the assignment.
      * @return Optional containing DoctorPatientAssignmentModel if found, empty otherwise.
      */
-    Optional<DoctorPatientAssignmentModel> findByDoctorIDAndPatientID(UUID doctorID, UUID patientID);
+    List<DoctorPatientAssignmentModel> findByDoctorIDAndPatientID(UUID doctorID, UUID patientID);
```

Comment:
To check if the doctor is already assigned to the patient, use this method **findByDoctorIdAndPatientIdAndD ateOfUnassignmentNotNull** . This will eliminate fetching unwanted records and increase performance. Also I suggest to refactor the DoctorPatientAssignmentModel's patientID and doctorID as patientId and doctorId respectively.

Reply:
Do we need to check scenarios such as whether the patient is assigned to a different doctor other than the doctorId specified in the request? Additionally, can a patient be assigned to more than one doctor at the same time?

Reply:
Changed the method name to findByDoctorIdAndPatientIdAndDateOfUnassignmentNull() , so that it would return only required records.